### PR TITLE
node: Mempool handling for moonlight txs

### PR DESCRIPTION
**Pull request by herr-seppia**
- Reuse the current mempool features for moonlight txs too
  - Check nonce repetition
  - Transaction replacement (according to gas price)
-  Don't discard transactions that are not ready yet (nonce is not the next one)

### CODE REVIEW

File: node-data/src/ledger/transaction.rs [hunk 1]
```diff
@@ -90,6 +97,24 @@ impl PartialEq<Self> for SpentTransaction {
 
 impl Eq for SpentTransaction {}
 
+pub enum SpendingId {
+    Nullifier([u8; 32]),
```
Comment (moCello) [inline node-data/src/ledger/transaction.rs:101]: In this implementation the `SpendingId` for phoenix would be only one nullifier. We can however have up to 4 in a transaction and the previous implementation of `to_nullifiers` also returned a `Vec<[u8; 32]>` (one nullifier is 32 bytes).

**Review by moCello — CHANGES_REQUESTED** [review]
Not all the nullifiers are included in the `SpendingId` for phoenix-transactions. For the rest it looks good!

File: node-data/src/ledger/transaction.rs (hunk 1 above)
Comment (herr-seppia) [reply node-data/src/ledger/transaction.rs:101]: Transaction::to_spending_ids indeed return a Vec<SpendingId>

So the idea is that a PhoenixTx has 4SpendingId to check, and MoonlightTx only one

File: node/src/mempool.rs [hunk 2]
```diff
@@ -192,23 +192,18 @@ impl MempoolSrv {
...
         db.read().await.update(|db| {
-            let nullifiers: Vec<_> = tx
-                .inner
-                .nullifiers()
-                .iter()
-                .map(|nullifier| nullifier.to_bytes())
-                .collect();
+            let spend_ids = tx.to_spend_ids();
 
             // ensure nullifiers do not exist in the mempool
```
Comment (moCello) [inline node/src/mempool.rs:197]: nit: comment not adjusted to spend-id's

**Review by moCello — APPROVED** [review]
LGTM

**Pull request merged**